app/models/service_container_template.rb

```diff
@@ -0,0 +1,104 @@
+class ServiceContainerTemplate < ServiceGeneric
+  delegate :container_template, :container_manager, :to => :service_template, :allow_nil => true
+
+  # A chance for taking options from automate script to override options from a service dialog
+  def preprocess(action, add_options = {})
+    unless add_options.blank?
+      _log.info("Override with new options:")
+      $log.log_hashes(add_options)
+    end
+
+    save_job_options(action, add_options)
+  end
+
+  def execute(action)
+    opts = get_job_options(action)
+
+    _log.info("Container template provisioning with options:")
+    $log.log_hashes(opts)
+
+    params = process_parameters(opts[:parameters])
+    ManageIQ::Providers::Openshift::ContainerManager::OrchestrationStack.create_stack(container_template, params, opts[:container_project_name])
+    _log.info("Provisioning of container template with ID: [#{id}] name:[#{name}] was initiated.")
+  end
+
+  def check_completed(action)
+    done = stack(action).provider_objects.all? { |obj| obj.resource_category.constantize.find_by(:name => obj.name) }
+    message = done ? "completed" : "not completed yet"
+    [done, message]
+  end
+
+  def refresh(action)
+  end
+
+  def check_refreshed(_action)
+    [true, nil]
+  end
+
+  def on_error(action)
+    _log.info("on_error called for service action: #{action}")
+    container_template.rollback_objects(@created_objects)
+  end
+
+  def stack(action)
+    service_resources.find_by(:name => action, :resource_type => 'OrchestrationStack').try(:resource)
+  end
+
+  private
+
+  def process_parameters(inputs)
+    params = container_template.container_template_parameters.to_a
+    inputs.each do |key, value|
+      match = params.find { |p| p.name == key.to_s }
+      match.value = value if match
+    end
+    params
+  end
+
+  def get_job_options(action)
+    options[job_option_key(action)].deep_dup
+  end
+
+  def save_job_options(action, overrides)
+    return if action == ResourceAction::RETIREMENT
+
+    job_options = {
+      :container_project_name => project_name(overrides),
+      :parameters             => parameters_from_dialog.with_indifferent_access.merge(overrides)
+    }
+
+    options[job_option_key(action)] = job_options
+    save!
+  end
+
+  def job_option_key(action)
+    "#{action.downcase}_job_options".to_sym
+  end
+
+  def parameters_from_dialog
+    params =
+      options[:dialog].each_with_object({}) do |(attr, val), obj|
+        var_key = attr.sub(/dialog_param_/, '')
+        obj[var_key] = val unless var_key == attr
+      end
+
+    params.blank? ? {} : params
+  end
+
+  def project_name(overrides)
+    # :dialog option should specify the project name, either an existing project or a new project name
+    dialog_options = options[:dialog]
+    existing = overrides.delete(:existing_project_name) || dialog_options['dialog_existing_project_name']
+    new = overrides.delete(:new_project_name) || dialog_options['dialog_new_project_name']
```

Comment:
@lfu Avoid using keywords as variable names and would also expect this to be a more descriptive name.  Suggest `new_project_name`